### PR TITLE
feat: [CDS-47282]: label as ReactElement for ThumbnailSelect component

### PR DESCRIPTION
**Pull request by Eric761**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Review by ishantmehtaharness — DISMISSED** [review]

File: packages/uicore/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@harness/uicore",
-  "version": "3.96.0",
+  "version": "3.96.1",
```
Comment (AshwinBhatkal-Harness) [inline packages/uicore/package.json:3]:
```suggestion
  "version": "3.97.0",
```

File: packages/uicore/src/components/Thumbnail/Thumbnail.tsx [hunk 2]
```diff
@@ -71,15 +71,19 @@ export const Thumbnail: React.FC<ThumbnailProps> = props => {
...
-          className={css.label}
-          font={{ weight: 'semi-bold' }}
-          color={disabled ? Color.GREY_500 : Color.GREY_600}
-          margin={{ top: 'small' }}>
-          {label}
-        </Text>
-      )}
+      {(icon || imageProps?.src) &&
+        label &&
+        (typeof label === 'string' ? (
```
Comment (AshwinBhatkal-Harness) [inline packages/uicore/src/components/Thumbnail/Thumbnail.tsx:76]: Why is this required?

File: packages/uicore/package.json (hunk 1 above)
Comment (Eric761) [reply packages/uicore/package.json]: done

File: packages/uicore/src/components/Thumbnail/Thumbnail.tsx (hunk 2 above)
Comment (Eric761) [reply packages/uicore/src/components/Thumbnail/Thumbnail.tsx]: If someone wants to use label as a ReactElement, then we don't need to wrap it around the Text component & instead of using typeof to check for ReactElement existence, we can use isValidElement.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
